cli_tools/common/vdisk/file_inspector.go

```diff
@@ -0,0 +1,104 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package vdisk
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"path"
+	"time"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/gcsfuse"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/files"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/storage"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+const bytesPerGB = int64(1024 * 1024 * 1024)
+
+// VirtualDiskFileMetadata contains metadata about a virtual disk file.
+type VirtualDiskFileMetadata struct {
+	// Gigabytes used by the file itself, rounded up to the nearest GB.
+	PhysicalSizeGB int64
+
+	// Gigabytes used by the disk, after inflation. Rounded up to the nearest GB.
+	VirtualSizeGB int64
+
+	// The file format used for encoding the VM disk.
+	FileFormat Format
+}
+
+// VirtualDiskFileInspector returns metadata about virtual disk files.
+type VirtualDiskFileInspector interface {
+	// Inspect returns VirtualDiskFileMetadata for the virtual disk file associated
+	// with a reference. IO operations will be retried until the context is cancelled.
+	Inspect(ctx context.Context, reference string) (VirtualDiskFileMetadata, error)
+}
+
+// NewGCSInspector returns a virtualDiskFileInspector that inspects virtual
+// disk files that are stored in the GCS bucket. The Inspect method expects
+// a GCS URI to the file to be inspected.
+func NewGCSInspector() VirtualDiskFileInspector {
+	return gcsInspector{
+		qemuClient: NewInfoClient(),
+		fuseClient: gcsfuse.NewClient()}
+}
+
+// gcsInspector implements virtualDiskFileInspector using qemu-img gcsfuse.
+type gcsInspector struct {
+	qemuClient InfoClient
+	fuseClient gcsfuse.Client
+}
+
+func (inspector gcsInspector) Inspect(ctx context.Context, gcsURI string) (metadata VirtualDiskFileMetadata, err error) {
```

Comment:
In the original plan, there would be a "inspection stage" after the "inflation stage". Now you do a inspection even earlier than the inflation. SO will there be 2 inspections in total? If so, we should name them carefully to avoid confusion.

Reply:
BTW I found a golang guestfs lib: http://libguestfs.org/guestfs-golang.3.html

Reply:
> In the original plan, there would be a "inspection stage" 

You're right, I was similarly less-than-happy with overloading the word "inspection". Given the naming proposal above, if we picked `imagefile.Inspector`, then at least the term "inspect" is associated with the image file.

Zooming out, I don't see this as an architectural component of inflation, but rather an implementation detail of the inflation stage. 

Any ideas for terms that'd be clearer?

Reply:
I think it's good to have image file inspector v.s. disk inspector to distinguish them. How do you think?

Reply:
Yeah, so this PR adds `imagefile.Inspector` -- sound okay?

Reply:
Yes~